### ISSUE TITLE
Issues with German umlauts 

### ISSUE THREAD

**Issue by HephaistosGit** [Humanizr/Humanizer#532]
Hello,

there are problems with special german letters:

(Humanizer v.2.0.1, Visual Studio 2013, .NET, 4.6.1)

```
string city = "MÜNCHEN";
Console.WriteLine("{0} => {1}", city, city.Humanize().Transform(To.LowerCase, To.TitleCase));

city = "münchen";
Console.WriteLine("{0} => {1}", city, city.Humanize().Transform(To.LowerCase, To.TitleCase));

city = "München";
Console.WriteLine("{0} => {1}", city, city.Humanize().Transform(To.LowerCase, To.TitleCase));
```

Output should be "München", is:

```
MÜNCHEN => München
münchen => M Nchen
München => Nchen
```

Any suggestions on how I can solve the problem?

**Comment by HephaistosGit** [comment]
Another challenge is the letter "ß" (for this one exists no capital letter)

```
string street = "Mühldorfer Straße 23";
Console.WriteLine("{0} => {1}", street, street.Humanize().Transform(To.LowerCase, To.TitleCase));

street = "MÜHLDORFER strAße 23";
Console.WriteLine("{0} => {1}", street, street.Humanize().Transform(To.LowerCase, To.TitleCase));

street = "mühldorfer STRAßE 23";
Console.WriteLine("{0} => {1}", street, street.Humanize().Transform(To.LowerCase, To.TitleCase));
```

```
Mühldorfer Straße 23 => Hldorfer Stra E 23
MÜHLDORFER strAße 23 => Hldorfer Str E 23
mühldorfer STRAßE 23 => M Hldorfer E 23
```

**Comment by aloisdg** [comment]
@HephaistosGit Could you add this to unit tests with a small PR?

**Comment by HephaistosGit** [comment]
I've added a tiny solution...

```
...
[Test]
public void HumanizeCityTest() {
   const string cityUppercase = "MÜNCHEN";
   const string cityLowercase = "münchen";
   const string cityNormal = "München";
   const string cityExpected = "München";

   // Tests: SUCCESS
   Assert.AreEqual(cityExpected, cityUppercase.ToUpper().Humanize().Transform(To.LowerCase, To.TitleCase));
   Assert.AreEqual(cityExpected, cityLowercase.ToUpper().Humanize().Transform(To.LowerCase, To.TitleCase));
   Assert.AreEqual(cityExpected, cityNormal.ToUpper().Humanize().Transform(To.LowerCase, To.TitleCase));

   // Tests: FAILED
   Assert.AreEqual(cityExpected, cityUppercase.Humanize().Transform(To.LowerCase, To.TitleCase));
   Assert.AreEqual(cityExpected, cityLowercase.Humanize().Transform(To.LowerCase, To.TitleCase));
   Assert.AreEqual(cityExpected, cityNormal.Humanize().Transform(To.LowerCase, To.TitleCase));
}

[Test]
public void HumanizeStreetTest() {
   const string streetUppercase = "MÜHLDORFER strAße 23";
   const string streetLowercase = "mühldorfer STRAßE 23";
   const string streetNormal = "Mühldorfer Straße 23";
   const string streetExpected = "Mühldorfer Straße 23";

   // Tests: Failed
   Assert.AreEqual(streetExpected, streetUppercase.ToUpper().Humanize().Transform(To.LowerCase, To.TitleCase));
   Assert.AreEqual(streetExpected, streetLowercase.ToUpper().Humanize().Transform(To.LowerCase, To.TitleCase));
   Assert.AreEqual(streetExpected, streetNormal.ToUpper().Humanize().Transform(To.LowerCase, To.TitleCase));
}
..
```

Since this is my first visit to GitHub, I beg your indulgence if I made mistakes.

[HumanizerTests.zip](https://github.com/Humanizr/Humanizer/files/137926/HumanizerTests.zip)

**Comment by aloisdg** [comment]
> Since this is my first visit to GitHub, I beg your indulgence if I made mistakes.

First time? :tada: 

Before anything thank for your Issue! Now lets see for your first Pull Request. First, go to [this page](http://makeapullrequest.com/). You will find a serie of screencast to help you. For pull request you can watch [this one](https://egghead.io/lessons/javascript-how-to-create-a-pull-request-on-github?series=how-to-contribute-to-an-open-source-project-on-github) in particular.

If you have any questions not directly related to your issue but to your pull request, feel free to come on the [chat](https://gitter.im/Humanizr/Humanizer/~chat#). I will be here.

**Comment by hangy** [comment]
This seems to be caused by [`PascalCaseWordPartsRegex`](https://github.com/Humanizr/Humanizer/blob/dev/src/Humanizer/StringHumanizeExtensions.cs#L17) being limited to A-Z, a-z and 0-9 when it comes to word characters. I think `\w` (which should be Unicode aware and should also match other diacritics like ē) is most likely not used, as it would also match `_`?

**Comment by hazzik** [comment]
Fixed by #564
